mitiq/tests/test_folding_pyquil.py

```diff
@@ -1,48 +1,112 @@
 import numpy as np
 import random
 from pyquil import Program
-from pyquil.gates import X, Y, Z
-from mitiq.folding_pyquil import local_folding, unitary_folding, sampling_stretcher, left_stretcher
+from pyquil.gates import X, Y, Z, CNOT
+from pyquil.unitary_tools import program_unitary
+from mitiq.folding_pyquil import (
+    local_folding,
+    unitary_folding,
+    sampling_stretcher,
+    left_stretcher,
+)
+from copy import deepcopy
+
+
+STRETCH_VALS = [1.0, 1.5, 2.0, 2.5, 3.0, 3.5, 4.0]
```

Comment:
For testing purposes might be good to change this to use not only integer and half-integer values and not to be evenly spaced, e.g.

```suggestion
STRETCH_VALS = [1.0, 1.3, 2.0, 2.6, 3.0, 3.5, 4.0]
```